### ISSUE TITLE
Using XofReader

### ISSUE THREAD

**Issue by samscott89** [RustCrypto/hashes#42]
Hi,
Am I using the `XofReader` (`Sha3XofReader`) incorrectly? I was assuming that each call to `XofReader::read(...)` would extend the previous values.

For example, the following:

```rust
extern crate digest;
extern crate sha3;

use digest::{Input, ExtendableOutput ,XofReader};
use sha3::Shake256;

fn main() {
    let mut hasher = Shake256::default();
    hasher.process(b"some nice randomness here");
    let mut xof = hasher.xof_result();

    let mut buf = [0; 4];

    for _ in 0..5 {
        xof.read(&mut buf);
        println!("{:?}", buf);
    }
}
```
Repeatedly returns the same values.
```
[27, 145, 10, 182]
[27, 145, 10, 182]
[27, 145, 10, 182]
[27, 145, 10, 182]
[27, 145, 10, 182]
```

(Sorry if this is better placed in https://github.com/RustCrypto/traits, I figured it's probably an implementation bug, unless I've done something stupid.)

**Comment by newpavlov** [comment]
Thank you for reporting this!

It looks like a bug in `Sha3XofReader` implementation. I'll try to fix it today or tomorrow.

**Comment by newpavlov** [comment]
This bug should be fixed in the sha3 v0.7.1.

**Comment by samscott89** [comment]
Awesome, works perfectly. Thanks!
